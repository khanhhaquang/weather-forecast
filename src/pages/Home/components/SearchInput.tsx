import { useEffect, useState } from 'react';
import { useDispatch } from 'react-redux';
import styled from 'styled-components';
import { useDebounce } from 'hooks';
import { getWeather } from '../actions';

const Input = styled.input`
  outline: none;
  border: 1px solid #f0f0f0;
  box-shadow: 0px 6px 6px 0px rgb(162 162 162);
  height: 24px;
  padding: 10px 50px;
  width: 100%;
  font-size: 16px;
  line-height: 24px;
  margin-bottom: 8px;
  &::placeholder {
    font-style: italic;
    opacity: 0.5;
  }
`;

const SearchInput: React.FC = () => {
  const [value, setValue] = useState<string>('');
  const debouncedSearchValue = useDebounce(value);
  const dispatch = useDispatch();

  useEffect(() => {
    if (debouncedSearchValue) {
      dispatch(getWeather(debouncedSearchValue));
    }
  }, [debouncedSearchValue]);

  return (
    <Input
      placeholder="Which city do you want to check ?"
      value={value}
      onChange={(e) => setValue(e.target.value)}
    />
  );
};

export default SearchInput;
