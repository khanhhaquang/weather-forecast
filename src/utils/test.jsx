/* eslint-disable import/no-unresolved */
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { render as rtlRender } from '@testing-library/react';
import rootReducer from 'pages/reducer';

function render(ui, { initialState, store = createStore(rootReducer, initialState) } = {}) {
  return {
    ...rtlRender(<Provider store={store}>{ui}</Provider>),
    // adding `store` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    store,
  };
}

export * from '@testing-library/react';
export { render };
