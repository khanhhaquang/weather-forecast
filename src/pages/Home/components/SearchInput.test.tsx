import { render, screen, fireEvent } from 'utils/test';
import SearchInput from './SearchInput';

describe('Home/SearchInput', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<SearchInput />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should change correctly', () => {
    render(<SearchInput />);
    const input = screen.getByTestId('search-input');

    fireEvent.change(input, { target: { value: 'foo' } });
    expect(input.value).toBe('foo');
  });
});
