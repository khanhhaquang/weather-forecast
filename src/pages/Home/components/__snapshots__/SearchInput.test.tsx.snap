// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home/SearchInput should render correctly 1`] = `
<DocumentFragment>
  <input
    class="sc-bdnxRM iFwihy"
    data-testid="search-input"
    placeholder="Which city do you want to check ?"
    value=""
  />
</DocumentFragment>
`;
